Modules/Libs/QuestieLib.lua

```diff
@@ -142,15 +141,15 @@ function QuestieLib:GetColoredQuestName(questId, showLevel, showState, blizzLike
     end
 
     if showState then
-        local isComplete = quest:IsComplete()
+        local isComplete = QuestieDB.IsComplete(questId)
```

Comment:
What's the difference between `QuestieDB.IsComplete(questId)` and `QuestieDB:GetQuest(questId).isComplete` a few lines below?

Reply:
The QuestieDB.IsComplete() function is embedded into the `quest` table that is returned from QuestieDB:GetQuest()... I.E. `QuestieDB:GetQuest(questId):IsComplete()`. The difference? Code wise, nothing. But since I'm not storing the entire `quest` table into a local var I can't use `quest:IsComplete()`. Make sense?

Reply:
Sorry... you asked about `QuestieDB:GetQuest(questId).isComplete`. That is a key that gets flagged true when all objectives are complete. It's designed to identify quests that should be complete but `:IsComplete()` returns zero or -1.

Reply:
My point was: Why calling the heavy `QuestieDB:GetQuest(questId).IsComplete` when you already called `QuestieDB.IsComplete(questId)` a few lines before?

Is there a difference of outcome? Can the one be „more complete“ than the other? Or is it just `boolean` vs. `number`?

Reply:
`QuestieDB.IsComplete(questId)` and `QuestieDB:GetQuest(questId):IsComplete()` are the same function. It only returns -1, 1 or 0 and it's based on what Blizzard returns. There are quite a few quests that return zero or -1 that should be returning 1... this is why we set the `QuestieDB:GetQuest(questId).isComplete` flag to true when all the objectives are complete - in short it ignores `IsComplete()` and forces a quest to be treated as complete even though `IsComplete()` returns something other than 1. Make sense?